sycl/include/CL/sycl/ONEAPI/sub_group.hpp

```diff
@@ -295,29 +295,70 @@ struct sub_group {
                         PI_INVALID_DEVICE);
 #endif
   }
-
+#ifdef __SYCL_DEVICE_ONLY__
+#ifdef __NVPTX__
   template <int N, typename T, access::address_space Space>
   sycl::detail::enable_if_t<
-      sycl::detail::sub_group::AcceptableForGlobalLoadStore<T, Space>::value &&
-          N != 1,
+      sycl::detail::sub_group::AcceptableForGlobalLoadStore<T, Space>::value,
       vec<T, N>>
   load(const multi_ptr<T, Space> src) const {
-#ifdef __SYCL_DEVICE_ONLY__
-#ifdef __NVPTX__
     vec<T, N> res;
     for (int i = 0; i < N; ++i) {
       res[i] = *(src.get() + i * get_max_local_range()[0] + get_local_id()[0]);
     }
     return res;
-#else
+  }
+#else  // __NVPTX__
+  template <int N, typename T, access::address_space Space>
+  sycl::detail::enable_if_t<
+      sycl::detail::sub_group::AcceptableForGlobalLoadStore<T, Space>::value &&
+          N != 1 && N != 3 && N != 16,
+      vec<T, N>>
+  load(const multi_ptr<T, Space> src) const {
     return sycl::detail::sub_group::load<N, T>(src);
-#endif // __NVPTX__
-#else
+  }
+
+  template <int N, typename T, access::address_space Space>
+  sycl::detail::enable_if_t<
+      sycl::detail::sub_group::AcceptableForGlobalLoadStore<T, Space>::value &&
+          N == 16,
+      vec<T, 16>>
+  load(const multi_ptr<T, Space> src) const {
+    return {sycl::detail::sub_group::load<8, T>(src),
+            sycl::detail::sub_group::load<8, T>(src +
+                                                8 * get_max_local_range()[0])};
+  }
+
+  template <int N, typename T, access::address_space Space>
+  sycl::detail::enable_if_t<
+      sycl::detail::sub_group::AcceptableForGlobalLoadStore<T, Space>::value &&
+          N == 3,
+      vec<T, 3>>
+  load(const multi_ptr<T, Space> src) const {
+    auto res = sycl::detail::sub_group::load<4, T>(src);
+    return {res.s2(), res.s1(), res.s0()};
```

Comment:
Is the order correct here?

AFAIU, for `N=16` case the order of elements is 0, 1, 2..., 15, isn't it?

Reply:
thank you. It is sort of misprint.